src/main/java/echo/storage/Storage.java

```diff
@@ -0,0 +1,139 @@
+package echo.storage;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.util.Scanner;
+import java.util.regex.Pattern;
+
+import echo.task.DeadlineTask;
+import echo.task.EventTask;
+import echo.task.TaskList;
+import echo.task.TodoTask;
+import echo.utils.EchoException;
+
+/**
+ * Storage deals with loading tasks from the file and saving tasks in the file.
+ */
+public class Storage {
+
+    /** File path containing the saved task list. */
+    private final String FILE_PATH;
+
+    /**
+     * Constructor for Storage class.
+     *
+     * @param filePath File path of the saved task list.
+     */
+    public Storage(String filePath) {
+        this.FILE_PATH = filePath;
+    }
+
+    /**
+     * Returns FILE_PATH.
+     *
+     * @return FILE_PATH.
+     */
+    public String filePath() {
+        return this.FILE_PATH;
+    }
+
+    /**
+     * Loads user save file from FILE_PATH.
+     *
+     * @return TaskList loaded from FILE_PATH.
+     *
+     * @throws FileNotFoundException If file at FILE_PATH does not exist.
+     * @throws EchoException If file at FILE_PATH is not formatted properly.
+     */
+    public TaskList load() throws FileNotFoundException, EchoException {
+        File file = new File(FILE_PATH);
+
+        Scanner s = new Scanner(file);
+
+        TaskList tasks = new TaskList();
+
+        // Directory that the file is located in.
+        File directory = file.getParentFile();
+
+        // Creates the directory if it does not exist.
+        if (!directory.exists()) {
+            directory.mkdir();
+        }
+
+        while (s.hasNext()) {
+            String line = s.nextLine();
+            try {
+                String[] splitVerticalBar = line.split(Pattern.quote(" | "));
+                String type = splitVerticalBar[0];
+                String desc = splitVerticalBar[2];
+
+                // Add task to tasks based on type.
```

Comment:
"Adds" would be more appropriate!

Reply:
Thanks for identifying the grammatical errors